src/ios/ListLibraryItems.m

```diff
@@ -112,46 +113,65 @@ - (void)uploadItem:(CDVInvokedUrlCommand *)command {
+                    PHAssetResourceRequestOptions * options = [PHAssetResourceRequestOptions new];
+                    [options setNetworkAccessAllowed: YES];
+                    [options setProgressHandler:^(double progress) {
+                        NSLog(@"progress %f", progress);
+                    }];
             
-            [[PHAssetResourceManager defaultManager] writeDataForAssetResource:resource toFile:mLocalTempURL options:options completionHandler:^(NSError * _Nullable aError) {
-                if (aError) {
-                    // cannot fetch asset
-                    NSString * message = [NSString stringWithFormat:@"Cannot fetch asset %@ (%@)", libraryId, [aError localizedDescription]];
-                    NSMutableDictionary * json = [NSMutableDictionary dictionaryWithObjects:@[@"-1", message, libraryId, uploadUrl] forKeys:@[@"code", @"message", @"source", @"target"]];
-                    [self returnUploadResult:NO payload:json command:command];
+                    [[PHAssetResourceManager defaultManager] writeDataForAssetResource:resource toFile:mLocalTempURL options:options completionHandler:^(NSError * _Nullable aError) {
+                        if (aError) {
+                            // cannot fetch asset
+                            NSString * message = [NSString stringWithFormat:@"Cannot fetch asset %@ (%@)", libraryId, [aError localizedDescription]];
+                            NSMutableDictionary * json = [NSMutableDictionary dictionaryWithObjects:@[@"-1", message, libraryId, uploadUrl] forKeys:@[@"code", @"message", @"source", @"target"]];
+                            [self returnUploadResult:NO payload:json command:command];
+                        } else {
+                            // create upload session
+                            NSURLSessionConfiguration * config = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:libraryId];
+                            [config setDiscretionary:YES]; // iOS will perform background upload when better suited
+                            [config setSessionSendsLaunchEvents:YES]; // launch app when upload finishes, calls "handleEventsForBackgroundURLSession" in AppDelegate
+        //                    NSURLSessionConfiguration * config = [NSURLSessionConfiguration defaultSessionConfiguration];
+                            NSURLSession * session = [NSURLSession sessionWithConfiguration:config delegate:self delegateQueue:[NSOperationQueue mainQueue]];
+
+                            NSMutableURLRequest * request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:uploadUrl]];
+                            [request setHTTPMethod: httpMethod];
+                            
+                            for(NSString * header in [headers allKeys]) {
+        //                        [headers setObject:[self getMimeTypeFromPath:resource.originalFilename] forKey:@"Content-Type"];
```

Comment:
Why don't we want the mime type anymore?

Reply:
You right I should keep it. It seems to work as we except though.

I took a look at the stack documentation: https://docs.cozy.io/en/cozy-stack/files/#post-filesdir-id_1

We should not pass the mimetype to que querystring (we actually do) and only pass it to the header.

Will make the change, thanks!